### PR TITLE
Update distributed_training.py

### PR DESCRIPTION
**Pull request by PingHGao**
Better example for DDP training

## Motivation

The previous version of `example/distributed_training.py` can not run DDP training correctly. After the discussion in #688，we provide a better example.

## Modification

Only `example/distributed_training.py` is modified, mainly  the dataloader. And I have done sufficient test.

## BC-breaking (Optional)

Nope

## Use cases (Optional)

Nope

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: examples/distributed_training.py [hunk 1]
```diff
@@ -41,6 +41,14 @@ def compute_metrics(self, results):
         return dict(accuracy=100 * total_correct / total_size)
 
 
+class IntToTensor:
+    def __call__(self, data):
+        return torch.tensor(data)
+
+    def __repr__(self):
+        return self.__class__.__name__ + '()'
+
```
Comment (C1rN09) [inline examples/distributed_training.py:50]: This target_transform is not necessary here. Maybe you encountered error caused by MMEngine's default `collate_fn`? Our new [documentation](https://github.com/open-mmlab/mmengine/pull/696/files) (which is still WIP) detailed on this and you may give some feedback :smile:

File: examples/distributed_training.py [hunk 2]
```diff
@@ -57,29 +65,37 @@ def parse_args():
...
-        dataset=torchvision.datasets.CIFAR10(
-            'data/cifar10',
-            train=False,
-            download=True,
-            transform=transforms.Compose(
-                [transforms.ToTensor(),
-                 transforms.Normalize(**norm_cfg)])))
+        dataset=valid_set,
+        sampler=dict(type='DefaultSampler', shuffle=True),
+        collate_fn=dict(type='default_collate'))
```
Comment (C1rN09) [inline examples/distributed_training.py:98]: I think in this example, it's better to remove `shuffle` parameter in  the outer `dict` because it might be misleading. And it might be better to change `shuffle` in `val_dataloader` to `False`, which is a common practice.

**Comment by PingHGao** [comment]
Thanks for your review, I have updated the file and tested. @C1rN09

**Review by C1rN09 — DISMISSED** [review]

**Comment by C1rN09** [comment]
Sorry I approved by mistake. Your lint failed and you can fix it by using `pre-commit hooks` following this [guide](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md#code-style)

Or to be simple, you can run these commands in mmengine project root directory

```shell
pip install -U pre-commit
pre-commit install
pre-commit run --all-files
```

**Comment by C1rN09** [comment]
I see you change it back to the original wrong version... Is it by mistake?

**Comment by PingHGao** [comment]
> I see you change it back to the original wrong version... Is it by mistake?

Sorry, I missed up the main branch. I think this update should be allright.

**Review by C1rN09 — APPROVED** [review]
